File: delightreading-server/src/main/java/com/delightreading/user/UserAccountRepository.java
package com.delightreading.user;

import com.delightreading.user.model.UserAccountEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserAccountRepository extends JpaRepository<UserAccountEntity, Long> {

    Optional<UserAccountEntity> findByUid(String uid);

    Optional<UserAccountEntity> findByUsername(String username);
}

File: delightreading-backend/src/main/kotlin/com/delightreading/user/UserAccountRepository.kt
package com.delightreading.user

import com.delightreading.user.model.UserAccountEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserAccountRepository : JpaRepository<UserAccountEntity, Long> {
    fun findByUid(uid: String): UserAccountEntity?

    fun findByUsername(username: String): UserAccountEntity?
}